maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -956,21 +979,7 @@ private Toolchain getToolchain() throws MojoFailureException
 
         if ( getJdkToolchain() != null )
         {
-            Method getToolchainsMethod = tryGetMethod( ToolchainManager.class, "getToolchains",
-                MavenSession.class, String.class, Map.class );
-            if ( getToolchainsMethod != null )
-            {
-                //noinspection unchecked
-                List<Toolchain> tcs = (List<Toolchain>) invokeMethodWithArray( getToolchainManager(),
-                    getToolchainsMethod, getSession(), "jdk", getJdkToolchain() );
-                if ( tcs.isEmpty() )
-                {
-                    throw new MojoFailureException(
-                        "Requested toolchain specification did not match any configured toolchain: "
-                            + getJdkToolchain() );
-                }
-                tc = tcs.get( 0 );
-            }
+            tc = getToolchainMaven33x( ToolchainManager.class, getToolchainManager(), getSession(), getJdkToolchain() );
```

Comment:
@akomakom 
We need to cover this line, and here the mocks would really help. Mock private static method and verifying that it was called and returns null and the fallback should be called:
`getToolchainManager().getToolchainFromBuildContext( "jdk", getSession() )`

Reply:
Done